synapse/handlers/identity.py

```diff
@@ -197,8 +197,22 @@ def try_unbind_threepid_with_id_server(self, mxid, threepid, id_server):
 
     @defer.inlineCallbacks
     def requestEmailToken(
-        self, id_server, email, client_secret, send_attempt, next_link=None
+        self, email, client_secret, send_attempt, next_link=None, **kwargs
     ):
+        """
+        Request an external server send an email on our behalf for the purposes of threepid
+        validation.
+
+        Args:
+            email (str): The email to send the message to
+            client_secret (str): The unique client_secret sends by the user
+            send_attempt (int): Which attempt this is
+            next_link: A link to redirect the user to once they submit the token
+            kwargs: extra arguments to send to the server
```

Comment:
a) we're not using kwargs, b) do we want to explicitly remove the `id_server` param here?

Do we really want to blindly pass through all args to the ID server anyway?

Reply:
> Do we really want to blindly pass through all args to the ID server anyway?

We already do so with the Msisdn endpoint, but indeed it does leave things open for potential issues (getting the homeserver to sign stuff and then send it off to the IS).

> do we want to explicitly remove the id_server param here?

Erm, you're probably right. We're still supporting this for the purposes of binding, where a user should be able to specify their IS. Will fix up.